src/Linux/local_cache.cpp

```diff
@@ -361,14 +380,18 @@ std::unique_ptr<std::vector<uint8_t>> local_cache_get(
 
     CacheEntryHeaderV1 header{};
     cache_file.read(reinterpret_cast<char*>(&header), sizeof(header));
-
-    if (checkExpiration && header.expiry <= time(nullptr))
+	
+    if (checkExpiration)
     {
-        cache_file.close();
-        unlink(file_name.c_str());
-        // Even if unlink fails, we can just return null. Thus, the return
-        // value is intentionally ignored here.
-        return nullptr;
+		last_cache_read_expiry = static_cast<long int> (header.expiry);
+		last_cache_read_time = static_cast<long int> (time(nullptr));
+		if(header.expiry <= time(nullptr)){
+			cache_file.close();
+			unlink(file_name.c_str());
+			// Even if unlink fails, we can just return null. Thus, the return
+			// value is intentionally ignored here.
+			return nullptr;
+		}

```

Comment:
Please consider configuring your dev environment so that formatting and tabbing/spacing is consistent with what is already on the codebase. This conventions make it easy to read code in the future.